grails-app/controllers/au/org/emii/netcdfsubset/SubsetController.groovy

```diff
@@ -21,13 +23,29 @@ class SubsetController {
         response.setContentType("application/octet-stream")
         response.setHeader("Content-disposition", "filename=${filenameToServe(subsetRequest)}")
 
-        try {
-            subsetService.subset(subsetRequest.typeName, subsetRequest.CQL_FILTER, response.outputStream)
+        if (isThreadAvailable()) {
```

Comment:
The issue that i have is that successive subset operations after the first don't trigger an interrupt on the running thread, instead they just queue up...